ui/ui-config/pages/index.tsx

```diff
@@ -1,47 +1,104 @@
+import { useRouter } from "next/router";
+import { useApi } from "../../ui-components/hooks/useApi";
 import Head from "next/head";
-import { Row, Col } from "react-bootstrap";
-import Link from "next/link";
-import { useApi } from "@grouparoo/ui-components/hooks/useApi";
-import { Actions } from "@grouparoo/ui-components/utils/apiData";
+import { Row, Col, Image, Button } from "react-bootstrap";
+import { useState, useEffect } from "react";
+import Loader from "../../ui-components/components/loader";
+
+export default function Page(props) {
+  const router = useRouter();
+  const { navigationMode, navigation } = props;
+  const [shouldRender, setShouldRender] = useState(false);
+  const [CTAs, setCTAs] = useState({
+    CTALink: "/session/sign-in",
+    CTAMessage: "Sign In",
+    CTATarget: null,
+  });
+  let currentStep = null;
+
+  useEffect(() => {
+    if (navigationMode === "config:authenticated") {
+      getSetupSteps(props);
+    } else {
+      setShouldRender(true);
+    }
+
+    async function getSetupSteps(props) {
+      console.log(`fetching steps`);
+      const { execApi } = useApi(props);
+      const { setupSteps } = await execApi("get", `/setupSteps`);
+      const foundStep = await setupSteps.find(
+        (step) => !step.complete && !step.skipped
+      );
+      currentStep = foundStep;
+
+      if (navigationMode === "config:authenticated" && currentStep) {
+        setCTAs({
+          CTAMessage: "Set Up Grouparoo",
+          CTALink: "/setup",
+          CTATarget: null,
+        });
+      } else if (navigationMode === "config:authenticated" && !currentStep) {
+        setCTAs({
+          CTAMessage: "Configure Profiles",
+          CTALink: "/profiles",
+          CTATarget: null,
+        });
+      }
+      setShouldRender(true);
+    }
+  }, []);
+
+  if (shouldRender === false) {
+    return <Loader />;
+  }
 
-export default function IndexPage() {
   return (
     <>
       <Head>
         <title>Grouparoo</title>
+        <p>NAV MODE: {navigationMode}</p>
+        <p>{JSON.stringify(CTAs, null, 2)}</p>
```

Comment:
This one is on me for adding these while we were debugging.

![](https://media.giphy.com/media/yMaLDA976YtUs/giphy.gif)

Reply:
🐙